sdk/resources/Microsoft.Azure.Management.Resource/src/Customizations/Models/GenericResource.cs

```diff
@@ -0,0 +1,30 @@
+namespace Microsoft.Azure.Management.ResourceManager.Models
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Resource information.
+    /// </summary>
+    public partial class GenericResource : Resource
+    {
+        /// <summary>
+        /// Initializes a new instance of the GenericResource class.
+        /// </summary>
+        /// <param name="id">Resource ID</param>
+        /// <param name="name">Resource name</param>
+        /// <param name="type">Resource type</param>
+        /// <param name="location">Resource location</param>
+        /// <param name="tags">Resource tags</param>
+        /// <param name="plan">The plan of the resource.</param>
+        /// <param name="properties">The resource properties.</param>
+        /// <param name="kind">The kind of the resource.</param>
+        /// <param name="managedBy">ID of the resource that manages this
+        /// resource.</param>
+        /// <param name="sku">The SKU of the resource.</param>
+        /// <param name="identity">The identity of the resource.</param>
+        public GenericResource(string id = default(string), string name = default(string), string type = default(string), string location = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), Plan plan = default(Plan), object properties = default(object), string kind = default(string), string managedBy = default(string), Sku sku = default(Sku), Identity identity = default(Identity))
```

Comment:
Not quite.  You want to have this signature, in which all parameters through the first changed parameter are required:

``` C#
public GenericResource(string id, string name, string type, string location, IDictionary<string, string> tags, Plan plan = default(Plan), object properties = default(object), string kind = default(string), string managedBy = default(string), Sku sku = default(Sku), Identity identity = default(Identity))
```
In this manner, any current code that uses the first 4 parameters will map to the new constructor; any existing code that uses tags or any parameter to the right of it will use this constructor.

Reply:
Good pt. Changed the signature.